imports/message-read-receipt/server/lib/ReadReceipt.js

```diff
@@ -81,6 +81,7 @@ export const ReadReceipt = {
 		return ReadReceipts.findByMessageId(message._id).map((receipt) => ({
 			...receipt,
 			user: Users.findOneById(receipt.userId, { fields: { username: 1, name: 1 } }),
```

Comment:
On the `getReceipts` method the `message` is passed by parameter. We can check the message object to ensure that the message came from a Livechat user or not. If the `message.token` exists, then you need to get the `guest` by finding it with the `LivechatVisitors` model, as you can see below:

```suggestion
		user:  message.token? LivechatVisitors.findById(receipt.userId, { fields: { username: 1, name: 1 } }) : Users.findOneById(receipt.userId, { fields: { username: 1, name: 1 } }) ,
```

Reply:
Ok, let me update that.